### PR TITLE
Issue 18132 site search not including all lang versions of same content

### PR DESCRIPTION
**Pull request by dsilvam**
Integration tests and being able to pass a site/host to the getEmployeeLikeContent in TestDataUtils

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java [hunk 1]
```diff
@@ -440,5 +448,117 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
...
+
+        boolean defaultContentToDefaultLangOriginalValue =
+                Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
+
+        try {
+
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", true);
+
+            final Host site = new SiteDataGen().nextPersisted();
+            Language lang1 = APILocator.getLanguageAPI().getDefaultLanguage();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java]: Codacy Issue found: [Local variable 'lang1' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5260866)

File: dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java [hunk 2]
```diff
@@ -440,5 +448,117 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
...
+
+            final String uuid = UUIDGenerator.generateUuid();
+
+            final Template template = new TemplateDataGen()
+                    .withContainer(container.getIdentifier(), uuid)
+                    .nextPersisted();
+
+            TemplateDataGen.publish(template);
+
+            HTMLPageAsset page = new HTMLPageDataGen(folder, template).languageId(lang1.getId())
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java]: Codacy Issue found: [Local variable 'page' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5260866)

File: dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java [hunk 3]
```diff
@@ -440,5 +448,117 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
...
+    /**
+     * Given sceneario: Multi-language content referenced from a page. Create a Site-Search run including all the languages of the content and
+     * check the two versions made it into the resulting index.
+     */
+
+    @Test
+    public void test_MultilangContent_IndexingAllLanguages()
+            throws DotPublishingException, JobExecutionException, DotDataException, IOException, DotSecurityException, WebAssetException {
+
+        boolean defaultContentToDefaultLangOriginalValue =
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java]: Codacy Issue found: [Local variable 'defaultContentToDefaultLangOriginalValue' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5260866)

File: dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java [hunk 4]
```diff
@@ -440,5 +448,117 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
...
+                Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
+
+        try {
+
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", true);
+
+            final Host site = new SiteDataGen().nextPersisted();
+            Language lang1 = APILocator.getLanguageAPI().getDefaultLanguage();
+            Language lang2 = new LanguageDataGen().nextPersisted();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java]: Codacy Issue found: [Local variable 'lang2' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5260866)

**Review by freddyucv — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**